src/main/java/org/adridadou/ethereum/propeller/EthereumFacade.java

```diff
@@ -478,10 +480,22 @@ public CompilationResult compile(SoliditySourceFile src) {
      * @param eventDefiniton Event definition that should be matched
      * @param address address of the smart contract that emits the events
      * @param optionalTopics Optional indexed event parameters, passed as 64 character hexidecimal string
-     * @return
+     * @param fromBlock From which block to search from for the events
+     * @param toBlock Latest block which should be searched from for the events
+     */
+    public List<EventData> getLogs(DefaultBlockParameter fromBlock, DefaultBlockParameter toBlock, SolidityEvent eventDefiniton, EthAddress address, String... optionalTopics) {
```

Comment:
What I would suggest here is to define fromBlock and toBlock as Optional<DefaultBlockParameter> and then define a default value there.

i.e. fromBlock.getOrElse(DefaultBlockParameterName.EARLIEST) and toBlock.getOrElse(DefaultBlockParameterName.LATEST)

Reply:
but still keep the overriden method I think that's good

Reply:
What is the benefit of having optional when already overloading the method? Just curious, I committed the change

Reply:
Would expect to or have optionals or have a overriden method

Reply:
then I propose Optional. This way it means that you can either specify a value or use the default one. Because I could imagine someone who wants from a certain point in time until the latest, or from the first block until a certain point and then you need to know about EARLIEST and LATEST.

So let's just use with Optional and not use the overriden method

Reply:
Agreed! Removed the overridden method